src/client/routes.tsx

```diff
@@ -34,101 +34,106 @@ export type RoutingConfig = {
 };
 
 export const GatewayRoutes = () => (
-  <Switch>
-    <Route exact path={Routes.SIGN_IN}>
-      <SignInPage />
-    </Route>
-    <Route exact path={Routes.REGISTRATION}>
-      <RegistrationPage />
-    </Route>
-    <Route exact path={`${Routes.REGISTRATION}${Routes.EMAIL_SENT}`}>
-      <RegistrationEmailSentPage />
-    </Route>
-    <Route exact path={Routes.RESET}>
-      <ResetPasswordPage />
-    </Route>
-    <Route exact path={`${Routes.RESET}${Routes.EMAIL_SENT}`}>
-      <EmailSentPage />
-    </Route>
-    <Route exact path={`${Routes.CHANGE_PASSWORD}${Routes.TOKEN_PARAM}`}>
-      <ChangePasswordPage />
-    </Route>
-    <Route path={`${Routes.PASSWORD}${Routes.RESET_CONFIRMATION}`}>
-      <ChangePasswordCompletePage />
-    </Route>
-    <Route exact path={`${Routes.RESET}${Routes.RESEND}`}>
-      <ResendPasswordPage />
-    </Route>
-    <Route exact path={`${Routes.RESET}${Routes.EXPIRED}`}>
-      <ResetPasswordSessionExpiredPage />
-    </Route>
-    <Route exact path={`${Routes.SET_PASSWORD}${Routes.RESEND}`}>
-      <SetPasswordResendPage />
-    </Route>
-    <Route exact path={`${Routes.SET_PASSWORD}${Routes.EXPIRED}`}>
-      <SetPasswordSessionExpiredPage />
-    </Route>
-    <Route path={`${Routes.SET_PASSWORD}${Routes.COMPLETE}`}>
-      <SetPasswordCompletePage />
-    </Route>
-    <Route path={`${Routes.SET_PASSWORD}${Routes.EMAIL_SENT}`}>
-      <EmailSentPage />
-    </Route>
-    <Route exact path={`${Routes.SET_PASSWORD}${Routes.TOKEN_PARAM}`}>
-      <SetPasswordPage />
-    </Route>
-    <Route exact path={`${Routes.CONSENTS}${Routes.CONSENTS_DATA}`}>
-      <ConsentsDataPage />
-    </Route>
-    <Route exact path={`${Routes.CONSENTS}${Routes.CONSENTS_COMMUNICATION}`}>
-      <ConsentsCommunicationPage />
-    </Route>
-    <Route exact path={`${Routes.CONSENTS}${Routes.CONSENTS_NEWSLETTERS}`}>
-      <ConsentsNewslettersPage />
-    </Route>
-    <Route exact path={`${Routes.CONSENTS}${Routes.CONSENTS_REVIEW}`}>
-      <ConsentsConfirmationPage />
-    </Route>
-    <Route exact path={`${Routes.WELCOME}${Routes.RESEND}`}>
-      <WelcomeResendPage />
-    </Route>
-    <Route exact path={`${Routes.WELCOME}${Routes.EXPIRED}`}>
-      <WelcomeSessionExpiredPage />
-    </Route>
-    <Route exact path={`${Routes.WELCOME}${Routes.EMAIL_SENT}`}>
-      <EmailSentPage />
-    </Route>
-    <Route exact path={`${Routes.WELCOME}${Routes.TOKEN_PARAM}`}>
-      <WelcomePage />
-    </Route>
+  <RouterRoutes>
+    <Route path={Routes.SIGN_IN} element={<SignInPage />} />
+    <Route path={Routes.REGISTRATION} element={<RegistrationPage />} />
```

Comment:
the `exact` prop has been removed from some of these. Does this break anything?

Reply:
From the upgrade guide https://reactrouter.com/docs/en/v6/upgrading/v5#upgrade-to-react-router-v6:

<Route exact> is gone. Instead, routes with descendant routes (defined in other components) use a trailing * in their path to indicate they match deeply

Reply:
So from that reasoning every route is "`exact`" by default, and you use a `*` if it's not exact.

In gateway every route is exact, so it won't break anything now by removing it.

Reply:
👍